frontend/src/components/core/StreamlitDialog/StreamlitDialog.tsx

```diff
@@ -160,23 +160,25 @@ function clearCacheDialog(props: ClearCacheProps): ReactElement {
   // component here but it's not working without it
   return (
     <HotKeys handlers={keyHandlers} attach={window}>
-      <Modal isOpen onClose={props.onClose}>
-        <ModalHeader>Clear Cache</ModalHeader>
-        <ModalBody>
-          <div>
-            Are you sure you want to clear the <code>@st.cache</code> function
-            cache?
-          </div>
-        </ModalBody>
-        <ModalFooter>
-          <ModalButton kind={Kind.SECONDARY} onClick={props.onClose}>
-            Cancel
-          </ModalButton>
-          <ModalButton kind={Kind.PRIMARY} onClick={props.confirmCallback}>
-            Clear cache
-          </ModalButton>
-        </ModalFooter>
-      </Modal>
+      <div data-testid="stClearCacheDialog">
+        <Modal isOpen onClose={props.onClose}>
```

Comment:
Is it possible to give the `data-testid` attribute to the `Modal` instead?

Reply:
<Modal> doesn't seem to pass these attributes down to the elements it creates, so I don't think so.